src/test/ODataClient.test.ts

```diff
@@ -0,0 +1,161 @@
+import * as path from "path";
+use(chaiAsPromised);
+
+describe("OData Client", () => {
+  const oDataClient: ODataClient = new ODataClient();
+  const oDataClientNewBase: ODataClient = new ODataClient("BASE");
+  let fetchStub: sinon.SinonStub;
+  let requestStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fetchStub = sinon.stub(ODataClient.prototype, "fetchJSON" as any);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    requestStub = sinon.stub(ODataClient.prototype, "createRequest" as any);
+    requestStub.returns("pass");
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it("Get OData report", async () => {
+    const returns = {
+      value: [1, 2],
+    };
+    fetchStub.resolves(returns);
+    let report = await oDataClient.getODataReport("-", "-");
+    expect(report.value.length).to.be.eq(2);
+    expect(report.value[0]).to.be.eq(1);
+    expect(fetchStub.calledWith(
+      "https://api.bentley.com/insights/reporting/odata/-",
+      "pass",
+    )).to.be.true;
+
+    report = await oDataClientNewBase.getODataReport("-", "-");
+    expect(fetchStub.calledWith(
+      "BASE/odata/-",
+      "pass",
+    )).to.be.true;
+  });
+
+  it("Get OData report metadata", async () => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fetchStub = sinon.stub(ODataClient.prototype, "fetchData" as any);
+    const request: RequestInit = {
+      body: "Test",
+    };
+    requestStub.returns(request);
+    const myOptions = { status: 200, statusText: "Test" };
+    let body: string = fs.readFileSync(path.join(__dirname, "test-data/validMetaData.xml"), "utf-8");
+    let response: Response = new Response(body, myOptions);
+    fetchStub.resolves(response);
+
+    let report = await oDataClient.getODataReportMetadata("-", "-");
+    expect(report).to.not.be.undefined;
+    expect(report[0].name).to.be.eq("EntityName");
+    expect(report[0].columns[0].name).to.be.eq("ECInstanceId");
+
+    body = fs.readFileSync(path.join(__dirname, "test-data/largeMetaData.xml"), "utf-8");
+    response = new Response(body, myOptions);
+    fetchStub.resolves(response);
+
+    report = await oDataClient.getODataReportMetadata("-", "-");
+    expect(report).to.not.be.undefined;
+    expect(report.length).to.be.eq(2);
+
+    myOptions.status = 400;
+    response = new Response(body, myOptions);
+    fetchStub.throws(response);
+    await expect(oDataClient.getODataReportMetadata("-", "-")).to.be.rejected;
+
+    expect(fetchStub.calledWith(
+      "https://api.bentley.com/insights/reporting/odata/-/$metadata",
+      request,
+    )).to.be.true;
+
+    myOptions.status = 200;
+    body = fs.readFileSync(path.join(__dirname, "test-data/emptyMetaData.xml"), "utf-8");
+    response = new Response(body, myOptions);
+    fetchStub.resolves(response);
+    report = await oDataClientNewBase.getODataReportMetadata("-", "-");
```

Comment:
There are no expectations for this last test. Add some.

Reply:
added an expectation for the returned value to be an empty array